app/graphql/types/query_type.rb

```diff
@@ -1,11 +1,16 @@
 class Types::QueryType < Types::BaseObject
   field :locations, [Types::LocationType] do
     description "MojoTech office locations"
+    argument :city_name, String, "City name of location", required: false
   end
 
-  def locations
-    Rails.cache.fetch('locations-response', expires_in: 15.minutes) do
-      Location.all
+  def locations(city_name: nil)
+    if city_name
+      Location.all.select { |l| l.city_name == city_name }
+    else
+      Rails.cache.fetch('locations-response', expires_in: 15.minutes) do
+        Location.all
+      end
```

Comment:
Should I be wrapping `Location.all.select { |l| l.city_name == city_name }`with the Rails.cache.fetch as well?

Reply:
Yes, but a probably a different key